docs/cog_guides/downloader.rst

```diff
@@ -0,0 +1,547 @@
+
+**Arguments**
+
+- ``<repo_name>`` The repo or repos to update all cogs from.
+- ``<revision>`` The revision to update to.
+- ``[cogs]`` The cog or cogs to update.
+
+.. _downloader-command-findcog:
+
+^^^^^^^
+findcog
+^^^^^^^
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]findcog <command_name>
+
+**Description**
+
+Find which cog a command comes from.
+
+This will only work with loaded cogs.
+
+Example:
+    - ``[p]findcog ping``
+
+**Arguments**
+
+- ``<command_name>`` The command to search for.
+
+.. _downloader-command-pipinstall:
+
+^^^^^^^^^^
+pipinstall
+^^^^^^^^^^
+
+.. note:: |owner-lock|
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]pipinstall [deps...]
+
+**Description**
+
+Install a group of dependencies using pip.
+
+Examples:
+    - ``[p]pipinstall bs4``
+    - ``[p]pipinstall py-cpuinfo psutil``
+
+Improper usage of this command can break your bot, be careful.
+
+**Arguments**
+
+- ``<deps>`` The package or packages you wish to install.
+
+.. _downloader-command-repo:
+
+^^^^
+repo
+^^^^
+
+.. note:: |owner-lock|
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repo 
+
+**Description**
+
+Base command for repository management.
+
+.. _downloader-command-repo-add:
+
+""""""""
+repo add
+""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repo add <name> <repo_url> [branch]
+
+**Description**
+
+Add a new repo.
+
+Examples:
+    - ``[p]repo add 26-Cogs https://github.com/Twentysix26/x26-Cogs``
+    - ``[p]repo add Laggrons-Dumb-Cogs https://github.com/retke/Laggrons-Dumb-Cogs v3``
+
+Repo names can only contain characters A-z, numbers, underscores, and hyphens.
+The branch will be the default branch if not specified.
+
+**Arguments**
+
+- ``<name>`` The name given to the repo.
+- ``<repo_url>`` URL to the cog branch. Usually GitHub or GitLab.
+- ``[branch]`` Optional branch to install cogs from.
+
+.. _downloader-command-repo-delete:
+
+"""""""""""
+repo delete
+"""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repo delete <repo_name>
+
+.. tip:: Aliases: ``repo remove``, ``repo del``
+
+**Description**
+
+Remove a repo and its files.
+
+Example:
+    - ``[p]repo delete 26-Cogs``
+
+**Arguments**
+
+- ``<repo_name>`` The name of an already added repo
+
+.. _downloader-command-repo-info:
+
+"""""""""
+repo info
+"""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repo info <repo_name>
+
+**Description**
+
+Show information about a repo.
+
+Example:
+    - ``[p]repo info 26-Cogs``
+
+**Arguments**
+
+- ``<repo_name>`` The name of the repo to show info about.
+
+.. _downloader-command-repo-list:
+
+"""""""""
+repo list
+"""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repo list 
+
+**Description**
+
+List all installed repos.
+
+.. _downloader-command-repo-update:
+
+"""""""""""
+repo update
+"""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repo update [repos...]
+
+**Description**
+
+Update all repos, or ones of your choosing.
```

Comment:
Maybe add a note that this does **not** update the installed cogs?

Reply:
Comment added below this line